stl/inc/xstring

```diff
@@ -2642,6 +2655,50 @@ private:
     }
 
 public:
+#if _HAS_CXX20
+    void _Move_assign_from_buffer(_Elem* const _Right, const size_type _Size, const size_type _Res) {
+        // Move assign from a buffer, used by basic_stringbuf
+        _Tidy_deallocate();
+        _Traits::assign(_Right[_Size], _Elem());
+        pointer _Fancy_right = pointer_traits<pointer>::pointer_to(*_Right);
+        auto& _My_data       = _Mypair._Myval2;
+        _My_data._Mysize     = _Size;
+        _My_data._Myres      = _Res;
+        if (_My_data._Large_string_engaged()) {
+            _Construct_in_place(_My_data._Bx._Ptr, _Fancy_right);
+        } else {
+            _Traits::copy(_My_data._Bx._Buf, _Right, _Res);
+            _Getal().deallocate(_Fancy_right, _Res);
+        }
+    }
+
+    struct _Released_buffer {
+        pointer _Ptr;
+        size_type _Size;
+        size_type _Res;
+    };
+
+    _NODISCARD _Released_buffer _Release_to_buffer(_Alloc& _Al) {
+        // Release to a buffer, or allocate a new one if in small string mode
+        auto& _My_data = _Mypair._Myval2;
+        if (_My_data._Large_string_engaged()) {
+            _Released_buffer _Result{_My_data._Bx._Ptr, _My_data._Mysize, _My_data._Myres};
+            _Tidy_leak();

```

Comment:
I would prefer to see the _Orphan_all in here explicitly rather than introducing another function since it helps make exception safety clearer. (Namely that we can't _Orphan_all until after the .allocate)